### DISCUSSION TITLE
Large negative solar value at 0:00 - 01:00

### DISCUSSION THREAD

**Discussion by IRJesoos** [macxq/foxess-ha#63] [Q&A] [answered]
I'm fairly new (two whole days) So this might be ignorance on my behalf, but both days have shown a large negative value between 00:00 and 01:00 each day, I've looked through the logs and see no probe from the cloud that has such a large negative value (two or three are -0.1 or so)
![image](https://user-images.githubusercontent.com/107948358/175259727-dfc757f1-0da1-4121-a63f-aab43942863b.png)
As such I start the day with a net negative solar production that eventually is surpassed as generating solar offsets it. My self-consumed is also boned but that's another issue.
![image](https://user-images.githubusercontent.com/107948358/175260215-a49fde52-2f1f-4308-904f-d49714f1bcff.png)

I've looked at other user's instances running and haven't seen this issue yet, so after any feedback/advice! Thanks all!

**Comment by Daibutt** [comment]
Not 100% sure but this looks like it might potentailly be a timing issue.
The dashboard looks at incremental changes in the energy sensors to work out hourly usage. If you are using daily_pv1_sum etc in the energy dashboard and that doesn't reset to zero till some time after 00:00 then it "thinks" you had 20 odd kWh at 00:00 and then at 01:00 you have zero - hence you've "used" 20kWh of PV between 00:00 and 01:00.
Not sure what might be causing that (timezone/daylight saving mix up in HA maybe).
One way round it could be to change your PV sensors in the dashboard to pv1_sum and pv2_sum. 
These should never reset so as far as the dashboard is concerned all you'll see is either zero or positive energy generation.
![image](https://user-images.githubusercontent.com/69276073/175270207-2eee3a4f-7e62-47df-9680-549dd0ae1a25.png)

**Reply by IRJesoos** [reply]
Hey Dailbutt, thanks for the response, your suggestion makes sense if it's incremental, I've updated my time to be a random +1 time to match BST - (GMT+01:00) Algiers). I'll see if that works as I don't seem to have pv1_sum or pv2_sum as an option for solar production on my energy dashboard. 
I'm guessing this is also the reason my self consumed energy solar is 0% each day as it starts negative and needs to offset that PLUS the consumed from grid too.

Thanks again!

**Reply by Daibutt** [reply]
No worries @IRJesoos, hope that fixes it for you. Algiers is a good choice for lots of sun anyway :)

**Comment by Daibutt** [comment]
@IRJesoos I forgot pv1_sum and pv2_sum etc. aren't part of the standard install.  I added these manually to take inverter losses out of the totals and also to understand each string's performance. 
Explanation and config info written up here;
https://github.com/macxq/foxess-ha/wiki/Understand-PV-string-power-generation-using-foxess-ha

**Reply by IRJesoos** [reply]
Unfortunately no luck on this one, being in Lagos still resulted in a the large offset this morning. I'm going for a forced shutdown and restart and will see what tomorrow brings as I found some resources stating some timezone changes needed that. If not I'll try the above if I can get my head around it!

![image](https://user-images.githubusercontent.com/107948358/175493453-7278b74a-ad75-4f8e-aba2-8f5e401fefd8.png)

**Comment by IRJesoos** [comment]
Just to add to this, a workaround to adjust the data from the first probe of the day can be done, but obviously isn't ideal. Still trying to resolve this. Changing timezones on the HA server has no impact, all times are correctly shown as BST time/date:
![image](https://user-images.githubusercontent.com/107948358/175808844-d9ee22cd-9d29-41c0-b7cf-62b358756d2d.png)

**Comment by IRJesoos** [comment, marked as answer]
Solution was an easy one. Energy Solar data input should be "FoxESS - Energy Generated" Not "FoxESS - Solar"

**Reply by r-amado** [reply]
I think this explanation is not correct. Energy Generated is the total amount of energy that the inverter gives to your network. Solar should be only the energy generated by the Solar Array. At a given time Energy Generated is the sum of Solar and Battery being injected in your network.
